DEVELOPERS.md

```diff
@@ -0,0 +1,421 @@
+# Developers Guide
+
+This guide is to assist developers who wish to modify one or more of the model components of ACCESS-OM2 and compile the modified code. Typically there is a development cycle, where the code is modified, compiled, tested and further modified based on testing.
+
+In particular this guide steps through the process to develop the ACCESS-OM2 model using [`spack`](spack.io), the build from source package manager that is use to build and deploy the model.
+
+It is assumed the modifications and compilation will be done on the [NCI HPC system `gadi`](https://opus.nci.org.au/display/Help/Gadi+User+Guide).
+
+Much of this adapted from the [spack Developer Workflows Tutorial](https://spack-tutorial.readthedocs.io/en/latest/tutorial_developer_workflows.html).
+
+## Initial set-up 
+
+The steps in this section need only be done once, as long as the versions of the three repositories are sufficient.
+
+It is necessary to make a local copy of `spack`, the configuration suitable for `gadi` (`spack-config`) and a package repository (`spack-packages`) which defines a number of the packages used by ACCESS-OM2.
+
+Adapted from [instructions on the forum](https://forum.access-hive.org.au/t/how-to-build-access-om2-on-gadi/1545)
+
+```bash
+git clone -c feature.manyFiles=true https://github.com/ACCESS-NRI/spack.git --branch releases/v0.21 --single-branch --depth=1
+git clone https://github.com/ACCESS-NRI/spack-packages.git --branch main
+git clone https://github.com/ACCESS-NRI/spack-config.git --branch main
+```
+
+Note: `spack` is cloned from the ACCESS-NRI fork as on occasion there are some fixes that are back-ported by ACCESS-NRI.
+
+Add all the `spack-config` settings to your local `spack` instance:
+```bash
+ln -s -r -v spack-config/v0.21/gadi/* spack/etc/spack/
+```
+
+## Enable spack
+
+For every new login or new shell it is necessary to add `spack` to your shell, as well as some other settings. 
+Do note the space between the `.` and the path, as we are not treating `spack-enable.bash` as an executable:
+```bash
+. spack-config/spack-enable.bash
+```
+
+## Development environment
+
+Spack [has environments](https://spack.readthedocs.io/en/latest/environments.html), which is similar to many other related implementations such as conda environments and python virtual environments.
+
+Environments create an isolated operating environment within which `spack` can only see and access packages that are added to the environment. In this repository the model to be built is defined using an environment in the `spack.yaml` file.
```

Comment:
```suggestion
Environments create an isolated operating environment within which `spack` can only see and access packages that are added to the environment. In this repository, the model to be built is defined using the `spack.yaml` environment file.
```

What repository are we talking about here? Is that the ACCESS-OM2 repository? This hasn't appeared in this document before but 3 other repositories have been introduced so it might be worth being explicit.
If `spack.yaml` is a file in the ACCESS-OM2 repo, add a link to it?

Reply:
Good point.